src/ui/screens/home/controller/VPNToggle.qml

```diff
@@ -19,7 +19,8 @@ MZButtonBase {
 
     function handleClick() {
         toolTip.close();
-        if (VPNController.state !== VPNController.StateOff) {
+        if (VPNController.state !== VPNController.StateOff &&
+            VPNController.state !== VPNController.StateOnPartial) {
```

Comment:
One scenario I think we'll need to resolve but not in this PR...
- We connect via the extension, we're in StateOnPartial...
- We connect via the client, now we're in StateOn...
- We deactivate from the client... we're in StateOff... but then that means Firefox is now also unprotected(?) 

This is how it is supposed to work from the PRD but I'm guessing we'll get foxfooder feedback saying this is unexpected behavior so we might think about how to persist the extension vpn state through a deactivation in the client.

Reply:
Given that the proxy api works on request basis, and we stream the state of the vpn we can simply request to re-enable the vpn to partial with the next request. Given we set a proxy in that case, the request would have a small delay. 
Or we check before disconnecting if there is an active web-extension connection. however then we rely on the port continuesly beeing open.